src/engraving/compat/midi/compatmidirenderinternal.cpp

```diff
@@ -1625,136 +1629,124 @@ void fillScoreVelocities(const Score* score, CompatMidiRendererInternal::Context
-                            velo.addHairpin(tick, etick, change, method, direction);
-                        }
+                if (change != 0) {
+                    Fraction etick = tick + d->velocityChangeLength();
+                    ChangeMethod method = ChangeMethod::NORMAL;
+                    for (track_idx_t i = d->staffIdx() * VOICES; i < (d->staffIdx() + 1) * VOICES; ++i) {
+                        context.velocitiesByTrack[i].addHairpin(tick, etick, change, method, direction);
                     }
+                }
+            }
+            break;
+            case VoiceAssignment::ALL_VOICE_IN_INSTRUMENT: {
+                Part* part = d->staff()->part();
+                staff_idx_t pStartStaff = part->staves().front()->idx();
+                staff_idx_t pEndStaff = part->staves().back()->idx();
+                if (d->staffIdx() < pStartStaff || d->staffIdx() > pEndStaff) {
                     break;
-                case DynamicRange::PART:
-                    if (dStaffIdx >= partStaff && dStaffIdx < partStaff + partStaves) {
-                        for (staff_idx_t i = partStaff; i < partStaff + partStaves; ++i) {
-                            Staff* stp = mainScore->staff(i);
-                            if (!stp->isPrimaryStaff()) {
-                                continue;
-                            }
-
-                            VelocityMap& stVelo = context.velocitiesByStaff[stp->idx()];
-                            stVelo.addDynamic(tick, v);
-                            if (change != 0) {
-                                Fraction etick = tick + d->velocityChangeLength();
-                                ChangeMethod method = ChangeMethod::NORMAL;
-                                stVelo.addHairpin(tick, etick, change, method, direction);
-                            }
-                        }
+                }
+                for (staff_idx_t i = pStartStaff; i <= pEndStaff; ++i) {
+                    Staff* stp = mainScore->staff(i);
+                    if (!stp->isPrimaryStaff()) {
+                        continue;
                     }
-                    break;
-                case DynamicRange::SYSTEM:
-                    for (size_t i = 0; i < mainScore->nstaves(); ++i) {
-                        Staff* sts = mainScore->staff(i);
-                        if (!sts->isPrimaryStaff()) {
-                            continue;
-                        }
-
-                        VelocityMap& stVelo = context.velocitiesByStaff[mainScore->staff(i)->idx()];
-                        stVelo.addDynamic(tick, v);
-                        if (change != 0) {
-                            Fraction etick = tick + d->velocityChangeLength();
-                            ChangeMethod method = ChangeMethod::NORMAL;
-                            stVelo.addHairpin(tick, etick, change, method, direction);
+                    for (track_idx_t i = stp->idx() * VOICES; i < (stp->idx() + 1) * VOICES; ++i) {
```

Comment:
This causes a compiler warning, because the name `i` is already used by the outer loop. Perhaps change the outer to `staffIdx` and the inner to `track`?